src/modules/jobs/controllers/pokemon.controller.ts

```diff
@@ -0,0 +1,35 @@
+import { Controller, Get, Query } from '@nestjs/common';
+
+import { PokemonService } from '../services/pokemon.service';
+
+@Controller('/pokemon')
+export class PokemonController {
+  constructor(private readonly pokemonService: PokemonService) {}
+
+  /**
+   * @description - route is used by frontend to get specific information on a pokemon
+   * @param {string} id - string representation of a specific pokemon's id number
+   * @param {string} name - the name of a pokemon
+   * @returns {boolean} - a JSON object that indicates failure on a bad request or the pokemon data on a successful request
+   */
+  @Get(['/'])
+  async getSpecificPokemon(@Query('id') id: string, @Query('name') name: string) {
+    await new Promise(resolve => setTimeout(resolve, 500)); // just adds a delay for feel so request isn't instantaneous.
+    
+
+    if (!this.pokemonService.isGenOne(id, name)) {
```

Comment:
This one is tripping me up a little. I will give it some thought but I think we need to do 1 of 2 things (we'll talk to tyler on this as well)

1. Move this logic inside of the service. for some reason calling this method here just doesn't seem right to me. not 100% sure just I guess based on the sort of philosophy of service files it seems like it should happen in the service file.

2. Change the name of this method to be more generic. maybe something like `validatePokemonData` or something like that. Then if we expand to more pokemon we don't need to come back here.

Whatever the case, looking at this, since it's technically 2 optional params as they're queries, maybe passing an object would be better. Idk the ideal way to set it up in terms of undefined and truthy checks and all that, but passing in like {id, name} and then the method can use what it needs.

We'll talk, nice job though!

Reply:
I think unless we want to validate the pokemon specifically with this logic, it is no longer needed. The try/catch you suggested we implement will handle any bad request submitted to the pokeapi/v2. The ONLY reason I could think to keep this is if we truly wanted to confine pokemon to a certain subset.